prow/github/client.go

```diff
@@ -2169,21 +2169,23 @@ func (c *client) GetFailedActionRunsByHeadBranch(org, repo, branchName, headSHA
 
 	var runs WorkflowRuns
 
-	url := url.URL{
+	u := url.URL{
 		Path: fmt.Sprintf("/repos/%s/%s/actions/runs", org, repo),
 	}
-	query := url.Query()
-
+	// setting the OR condition to get both PR and PR target workflows
+	// query.Encode() will escape the + sign to %2B but GitHub API doesn't accept it,
+	// instead it should be +, so we need to set the query manually to construct the URL like:
+	// /repos/org/repo/actions/runs?branch=branchName\u0026event=pull_request+OR+pull_request_target\u0026status=failure,
+	u.RawQuery = "event=pull_request+OR+pull_request_target"
```

Comment:
The `+` symbol is the result when encoding the space char. I guess this is why in GH docs they include it this way. So, just doing `query.Add("event", "pull_request OR pull_request_target")` it will be translated to `event=pull_request+OR+pull_request_target`. Therefore, there is no need for a manual integration here.

Reply:
thats a good idea, i will test it.